app/views/devise/sessions/new.html.erb

```diff
@@ -0,0 +1,11 @@
+<% title "Sign in" %>
+
+<%= form_for(resource_name, resource, :url => session_path(resource_name)) do |f| %>
+  <p><%= f.label :username %><br />
+  <%= f.text_field :username %></p>
```

Comment:
Do we need to permit `username` in devise params?

Reply:
done.